ffi module cleanup: context.h to frameobject.h (#1341)
PyO3/pyo3: CHANGELOG.md

and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [Unreleased]
### Changed

> - Deprecate FFI definition `PyCoro_Check` and `PyAsyncGen_Check` in favor of `PyCoro_CheckExact` and `PyAsyncGen_CheckExact` respectively, as these are the correct name in the Python headers. [#1341](https://github.com/PyO3/pyo3/pull/1341)

Reviewer: 👍 changelog message looks good, but I don't see the `#[deprecated]` attribute in `genobject.rs`?

```suggestion
- Deprecate FFI definitions `PyCoro_Check` and `PyAsyncGen_Check` in favor of `PyCoro_CheckExact` and `PyAsyncGen_CheckExact` respectively (the former are not part of the Python API). [#1341](https://github.com/PyO3/pyo3/pull/1341)
```

Author: Oops, I accidentally staged this. Removing.

> - Deprecate FFI definition `PyCoroWrapper_Check` and `PyGetSetDef_DICT` which have never been defined in the Python headers. [#1341](https://github.com/PyO3/pyo3/pull/1341)

Reviewer: Maybe also `PyGetSetDef_INIT`?

> - Deprecate FFI definition `PyImport_Cleanup`, which was removed in 3.9 and previously marked "for internal use only". [#1341](https://github.com/PyO3/pyo3/pull/1341)
> - Deprecate FFI definition `PyOS_InitInterrupts`, which was removed in 3.10 and previously undocumented. [#1341](https://github.com/PyO3/pyo3/pull/1341)

Reviewer: I think the detail here is slightly too much for a CHANGELOG (users can find out more from the PR link)

```suggestion
- Deprecate FFI definition `PyImport_Cleanup`  (removed in Python 3.9). [#1341](https://github.com/PyO3/pyo3/pull/1341)
- Deprecate FFI definition `PyOS_InitInterrupts` (removed in Python 3.10). [#1341](https://github.com/PyO3/pyo3/pull/1341)
```

> - Deprecate FFI definition `PyOS_AfterFork`; introduce `PyOS_BeforeFork`, `PyOS_AfterFork_Parent`, `PyOS_AfterFork_Child` when building for Python 3.7. [#1341](https://github.com/PyO3/pyo3/pull/1341)

Reviewer: The new methods should be in the `Added` section please :)

> - Deprecate FFI definitions `PyEval_CallObjectWithKeywords`, `PyEval_CallObject`, `PyEval_CallFunction`, `PyEval_CallMethod` when building for Python 3.9. [#1338](https://github.com/PyO3/pyo3/pull/1338)

Reviewer: Usually within the subsections we order by the individual PRs, so would prefer if this one is at the top of `Changed` please!
